sonar-application/src/test/java/org/sonar/application/ClusterTest.java

```diff
@@ -0,0 +1,144 @@
+    ).isEqualTo(false);
+  }
+
+  @Test
+  public void test_interface() throws SocketException {
+    Props props = initDefaultProps();
+    String ipAddress = findIPv4Address().getHostAddress();
+
+    if (ipAddress != null) {
+      props.set(ClusterProperties.INTERFACES, ipAddress);
+      props.set(ClusterProperties.PORT, "64000");
+
+      Cluster cluster = new Cluster(props);
+      assertThat(
+        cluster.hazelcastInstance.getConfig().getNetworkConfig().getInterfaces().isEnabled()
+      ).isEqualTo(true);
+      assertThat(
+        cluster.hazelcastInstance.getConfig().getNetworkConfig().getInterfaces().getInterfaces()
+      ).containsExactly(ipAddress);
+      assertThat(
+        ((InetSocketAddress) cluster.hazelcastInstance.getLocalEndpoint().getSocketAddress()).getPort()
+      ).isEqualTo(64000);
+      assertThat(
+        ((InetSocketAddress) cluster.hazelcastInstance.getLocalEndpoint().getSocketAddress()).getAddress().getHostAddress()
+      ).isEqualTo(ipAddress);
+    }
+  }
+
+  @Test
+  public void test_with_already_used_port() throws IOException {
+    Props props = initDefaultProps();
+    InetAddress ipAddress = findIPv4Address();
+
+    if (ipAddress != null) {
+      ServerSocket socket = null;
+      try {
+        socket = new ServerSocket(0, 100, ipAddress);
+        props.set(ClusterProperties.INTERFACES, ipAddress.getHostAddress());
+        props.set(ClusterProperties.PORT, "" + socket.getLocalPort());
+        expectedException.expect(HazelcastException.class);
+        Cluster cluster = new Cluster(props);
+      } finally {
+        if (socket != null) {
+          socket.close();
+        }
+      }
+    }
+  }
+
+  @Test
+  public void test_adding_port_to_members() {
+    Props props = initDefaultProps();
+    props.set(ClusterProperties.MEMBERS, "192.168.1.30");
+    Cluster cluster = new Cluster(props);
+
+    assertThat(
+      cluster.hazelcastInstance.getConfig().getNetworkConfig().getJoin().getTcpIpConfig().getMembers()
+    ).containsExactly("192.168.1.30:9003");
+  }
+
+  private Props initDefaultProps() {
+    Props props = new Props(new Properties());
+    props.set(ClusterProperties.ENABLED, "true");
+    ClusterProperties.completeDefaults(props);
+    return props;
+  }
+
+  private InetAddress findIPv4Address() throws SocketException {
+    Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+
+    while (interfaces.hasMoreElements()) {
+      NetworkInterface networkInterface = interfaces.nextElement();
+      if (!networkInterface.isVirtual()
+        && !networkInterface.isLoopback()
+        && !networkInterface.getName().startsWith("docker")) {
+        Enumeration<InetAddress> ips = networkInterface.getInetAddresses();
+        while (ips.hasMoreElements()) {
+          InetAddress ip = ips.nextElement();
+
+          if (ip instanceof Inet4Address) {
+            return ip;
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+}
```

Comment:
missing trailing line return

you may want to configure your editor to do that for you (have you applied https://github.com/SonarSource/sonar-developer-toolset?)